pkg/vulnerabilityreport/ttl_report_test.go

```diff
@@ -0,0 +1,126 @@
+package vulnerabilityreport_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/aquasecurity/trivy-operator/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/trivy-operator/pkg/ext"
+	"github.com/aquasecurity/trivy-operator/pkg/operator/etc"
+	"github.com/aquasecurity/trivy-operator/pkg/trivyoperator"
+	"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport"
+)
+
+const (
+	vreport = "vreport"
+	ns      = "default"
+)
+
+func TestRegenerateReportIfExpired(t *testing.T) {
+	// clock
+	clock := ext.NewSystemClock()
+
+	// scheme
+	scheme := trivyoperator.NewScheme()
+
+	// set the VulnerabilityScannerReportTTL
+	config, err := etc.GetOperatorConfig()
+	require.NoError(t, err)
+	hours, err := time.ParseDuration("24h")
+	require.NoError(t, err)
+	config.VulnerabilityScannerReportTTL = &hours
+
+	// logger object
+	logger := log.Log.WithName("testing")
+
+	// create TTLReport controller
+	instance := vulnerabilityreport.TTLReportReconciler{Logger: logger, Config: config, Clock: clock}
+
+	// vuln report data
+	vulnReport := v1alpha1.VulnerabilityReport{}
+	vulnReport.Name = vreport
+	vulnReport.Namespace = ns
+
+	tests := []struct {
+		name                  string
+		reportUpdateTimestamp time.Duration
+		wantError             bool
+		wantReportDeleted     bool
+		ttlStr                string
+		invalidReportName     bool
+	}{
+		{
+			name:                  "Report timestamp < TTL",
+			reportUpdateTimestamp: -15 * time.Hour, // < 24h TTL
+			wantError:             false,
+			wantReportDeleted:     false,
+			ttlStr:                "24h",
+		},
+		{
+			name:                  "Report timestamp exceeds TTL",
+			reportUpdateTimestamp: -25 * time.Hour, // > 24 TTL
+			wantError:             false,
+			wantReportDeleted:     true, // = time.Duration(0)
+			ttlStr:                "24h",
+		},
+		{
+			name:      "missing TTL annotation in the report",
+			wantError: false, // Ignoring report without TTL set
+			ttlStr:    "24h",
+		},
+		{
+			name:      "invalid TTL in the annotation",
+			ttlStr:    "badtime",
+			wantError: true,
+		},
+		{
+			name:              "non-existent report name",
+			invalidReportName: true,  // sets the report name to empty string
+			wantError:         false, // missing/invalid report ignored
+			wantReportDeleted: false,
```

Comment:
```suggestion
```

No meaning in expecting a delete of something that doesn't exist.

Reply:
removed